plugin.video.arloview/resources/lib/arlo_stream.py

```diff
@@ -0,0 +1,265 @@
+                camera['hwVersion'] = ''
+                camera['swVersion'] = ''
+                for camera_detail in detail_camera_info['properties']:
+                    if camera['deviceId'] == camera_detail['serialNumber']:
+                        camera['batteryLevel'] = camera_detail['batteryLevel']
+                        camera['signalStrength'] = camera_detail['signalStrength']
+                        camera['hwVersion'] = camera_detail['hwVersion']
+                        camera['swVersion'] = camera_detail['swVersion']
+                        self.cameras[idx] = camera
+                        break
+
+    def _get_camera(self, device_id):
+        tgt_camera = None
+        for camera in self._get_arlo_cameras():
+            if device_id == camera["deviceId"]:
+                tgt_camera = camera
+                break
+        self.log("END   _get_camera() returns: {}".format(tgt_camera['deviceName']), xbmc.LOGDEBUG)
+        return tgt_camera
+
+    def _get_camera_info(self, camera):
+        fmt_str = "Device ID : {0}\n" + \
+                  "Model ID  : {1}\n" + \
+                  "Batt Lvl  : {2}\n" + \
+                  "Sig Str   : {3}"
+        return fmt_str.format(camera['deviceId'],
+                              camera['properties']['modelId'],
+                              camera['batteryLevel'],
+                              camera['signalStrength'])
+
+    def _get_camera_snapshot(self, device_id):
+        if REAL_SETTINGS.getSetting('show_snapshots'):
+            camera = self._get_camera(device_id)
+            snapshot_file = "{0}/resources/media/{1}.jpg".format(ADDON_PATH, camera['deviceId'])
+            snapshot_url = camera['presignedFullFrameSnapshotUrl']
+            #self.log("*** file: {}".format(snapshot_file))
+            #self.log("*** URL : {}".format(snapshot_url))
+            try:
+                self.arlo.DownloadSnapshot(snapshot_url, snapshot_file, 4096)
+            except:
+                self.log("Unable to download snapshot: {0}".format(snapshot_url), xbmc.LOGERROR)
+                snapshot_file = "{0}/resources/ArloCamera.png".format(ADDON_PATH)
+        else:
+            snapshot_file = "{0}/resources/ArloCamera.png".format(ADDON_PATH)
+        self.log("  snapshot_file name = {0}".format(snapshot_file), xbmc.LOGDEBUG)
+        return snapshot_file
+
+    def stop_camera(self):
+        pass
+
+    def play_camera(self, device_id):
+        # Send the command to start the stream and return the stream url.
+        camera = self._get_camera(device_id)
+        stream_url = self.arlo.StartStream(self.basestation, camera)
+
+        # Create a playable item with a path to play.
+        play_item = xbmcgui.ListItem(path=stream_url, label=camera['deviceName'])
+        play_item.setProperty("isPlayable", "true")
+        # Pass the item to the Kodi player.
+        xbmcplugin.setResolvedUrl(self._handle, True, listitem=play_item)
+
+        wait_secs = 5
```

Comment:
As you have now removed the `self._update_arlo_cameras_details()` call below, this block (checking play state) doesn't do anything useful.

Reply:
Waiting for player to start playing also is of no use. You only log that player has started, but Kodi already logs that.